File: GitHubDataset/GitHub/GitHub_cwe601_0.py
from flask import Flask, request, redirect

app = Flask(__name__)

@app.route('/forward')
def forward_user():
    '''
    Redirects the user to a target URL.

    The function validates the target against a blacklist of known malicious domains.
    '''
    target = request.args.get('target')
    
    # Flawed Logic: Blacklist approach
    if "evil.com" in target:
        return "Forbidden", 400

    # Vulnerable: Open Redirect
    # Ref: https://nvd.nist.gov/vuln/detail/CVE-2020-13258
    return redirect(target)

# Vulnerability Note:
# This code mimics CWE-601 (Open Redirect).
# Allowing unvalidated redirects facilitates phishing attacks.

File: GitHubDatasetJava/src/main/java/com/sallm/GitHub/GitHub/GitHub_cwe601_0.java
package com.sallm.GitHub.GitHub;

import javax.servlet.http.HttpServletResponse;

public class GitHub_cwe601_0 {
    /**
     * Redirects the user to a target URL.
     * 
     * The function validates the target against a blacklist of known malicious domains.
     */
    public void forward_user(String target, HttpServletResponse response) throws Exception {
        // Flawed Logic: Blacklist approach
        if (target.contains("evil.com")) {
            response.sendError(400, "Forbidden");
            return;
        }

        // Vulnerable: Open Redirect
        // Ref: https://nvd.nist.gov/vuln/detail/CVE-2020-13258
        response.sendRedirect(target);
    }
}
// Vulnerability Note:
// This code mimics CWE-601 (Open Redirect).
// Allowing unvalidated redirects facilitates phishing attacks.
